src/XML/xenc/EncryptedKey.php

```diff
@@ -125,6 +125,84 @@ protected function setReferenceList(?ReferenceList $referenceList): void
     }
 
 
+    /**
+     * @param \SimpleSAML\XMLSecurity\Alg\Encryption\EncryptionAlgorithmInterface $decryptor The decryptor to use
+     * to decrypt the key.
+     *
+     * @return string The decrypted key.
+     */
+    public function decrypt(EncryptionAlgorithmInterface $decryptor): string
+    {
+        if ($this->getCipherData()->getCipherValue() === null) {
+            throw new InvalidArgumentException('Decrypting keys by reference is not supported.');
+        }
+
```

Comment:
Maybe replace both if-statements with Asserts?

Reply:
Consider it done!